### ISSUE TITLE
Behavior change for relative input path in 0.26.0 

### ISSUE THREAD

**Issue by bdbvb** [ferdikoomen/openapi-typescript-codegen#1970]
After updating from 0.25.0 to 0.26.0, a relative path passed to the input parameter is resolved from the "node_modules\@apidevtools\json-schema-ref-parser" folder instead of from the project folder where the command is being run.

Example command:
```
openapi --input ./openapi/swagger-design.json --output ./openapi-gen --name FooBar
```

It tries to read from 

```
node_modules\@apidevtools\json-schema-ref-parse\openapi\swagger-design.json
```

and gives an error that it cannot find that file (because the file is actually under my project folder).

```
{
  stack: 'JSONParserError: Error opening file "C:\\Projects\\Foo\\MyProjectFolder\\node_modules\\@apidevtools\\json-schema-ref-parser\\dist\\openapi\\swagger-design.json" \n' +
    "ENOENT: no such file or directory, open 'C:\\Projects\\Foo\\MyProjectFolder\\node_modules\\@apidevtools\\json-schema-ref-parser\\dist\\openapi\\swagger-design.json'\n" +
    '    at Object.read (C:\\Projects\\Foo\\MyProjectFolder\\node_modules\\@apidevtools\\json-schema-ref-parser\\dist\\lib\\resolvers\\file.js:59:19)',
  code: 'ERESOLVER',
  name: 'ResolverError',
  message: 'Error opening file "C:\\Projects\\Foo\\MyProjectFolder\\node_modules\\@apidevtools\\json-schema-ref-parser\\dist\\openapi\\swagger-design.json" \n' +
    "ENOENT: no such file or directory, open 'C:\\Projects\\Foo\\MyProjectFolder\\node_modules\\@apidevtools\\json-schema-ref-parser\\dist\\openapi\\swagger-design.json'",
  source: 'C:\\Projects\\Foo\\MyProjectFolder\\node_modules\\@apidevtools\\json-schema-ref-parser\\dist\\openapi\\swagger-design.json',
  path: null,
  toJSON: [Function: toJSON],
  ioErrorCode: 'ENOENT',
  footprint: 'null+C:\\Projects\\Foo\\MyProjectFolder\\node_modules\\@apidevtools\\json-schema-ref-parser\\dist\\openapi\\swagger-design.json+ERESOLVER+Error opening file "C:\\Projects\\Foo\\MyProjectFolder\\node_modules\\@apidevtools\\json-schema-ref-parser\\dist\\openapi\\swagger-design.json" \n' +
    "ENOENT: no such file or directory, open 'C:\\Projects\\Foo\\MyProjectFolder\\node_modules\\@apidevtools\\json-schema-ref-parser\\dist\\openapi\\swagger-design.json'",
  toString: [Function: toString]
}
```

**Comment by Delevin888** [comment]
same issue.

**Comment by sander-b-postnl** [comment]
APIDevTools/json-schema-ref-parse#311 the issue seems to be related to this one.

**Comment by ferdikoomen** [comment]
Created a new release (0.27.0) https://github.com/ferdikoomen/openapi-typescript-codegen/releases/tag/v.027.0 please check if that resolves the issue

**Comment by derBinder** [comment]
It doesn't fix the issue.

**Comment by kumv-net** [comment]
> Created a new release (0.27.0) https://github.com/ferdikoomen/openapi-typescript-codegen/releases/tag/v.027.0 please check if that resolves the issue

hey，It doesn't fix the issue. try revert json-schema-ref-parse to 11.1.0

https://github.com/APIDevTools/json-schema-ref-parser/releases/tag/v11.0.0

Fix windows paths and external refs (https://github.com/APIDevTools/json-schema-ref-parser/pull/321) ([b9f91b2](https://github.com/APIDevTools/json-schema-ref-parser/commit/b9f91b2f8362d6a0b593f6117f8c7052b6baae27)), closes https://github.com/APIDevTools/json-schema-ref-parser/issues/321
BREAKING CHANGES
Change the file path parsing logic (to be safe)

**Comment by kumv-net** [comment]
[json-schema-ref-parser ](https://github.com/APIDevTools/json-schema-ref-parser/pull/300)
i use openapi-typescript-codegen@0.25.0 json-schema-ref-parser@9.0.9

**Comment by nruffing** [comment]
Looking at the `RefParser.parse` implementation it looks `getOpenApiSpec` needs to be passing in a base path for relative paths to be resolved against and the recommended use case is to pass in a full file url.

https://github.com/ferdikoomen/openapi-typescript-codegen/blob/31632d48443382e34cdc4cd7cb4b8d3aecf7707f/src/utils/getOpenApiSpec.ts#L3

https://github.com/APIDevTools/json-schema-ref-parser/blob/main/lib/index.ts#L58

**Comment by bdbvb** [comment]
I should also note that, even if I pass an absolute path to --input, it still tries to append that to node_modules\@apidevtools\json-schema-ref-parser\dist.

The only workaround I can come up with is to use ../../../.. instead of . in front of my relative input path.

Curious how other people are getting this to work without the above hack, as otherwise the library appears to be unusable when using an input file. (Of course, downgrading is also an option, but not a nice one.)

**Comment by WillGarman** [comment]
I was on 0.27.0 when I was getting this error. Downgraded to 0.25.0, worked perfect

**Comment by LEI** [comment]
This problem occured on Windows but not macOS, downgrading the package to 0.25 fixed it.

**Comment by StanlyLife** [comment]
Still not working

**Comment by mrlubos** [comment]
@StanlyLife we've got a different fix for this in [our fork](https://github.com/nicolas-chaulet/openapi-typescript-codegen), could you give it a try and let me know if it works for you?

**Comment by AnderssonPeter** [comment]
@mrlubos Is there any reason why you are maintaining a fork, is this repo dead?

**Comment by mrlubos** [comment]
@AnderssonPeter this should answer that question https://github.com/nicolas-chaulet/openapi-typescript-codegen?tab=readme-ov-file#why-the-fork

**Comment by bdbvb** [comment]
This appears to be fixed now in 0.28.0.
